alg/viewshed.cpp

```diff
@@ -280,12 +282,19 @@ bool Viewshed::calcOutputExtent(int nX, int nY)
     oOutExtent.xStop = GDALGetRasterBandXSize(pSrcBand);
     oOutExtent.yStop = GDALGetRasterBandYSize(pSrcBand);
 
-    if (nX < 0 || nX >= oOutExtent.xStop || nY < 0 || nY >= oOutExtent.yStop)
+    if (!oOutExtent.containsY(nY))
     {
         CPLError(CE_Failure, CPLE_AppDefined,
-                 "The observer location falls outside of the DEM area");
+                 "Observer position above or below the raster "
+                 "not currently supported");
         return false;
     }
+    if (!oOutExtent.contains(nX, nY))
+    {
+        CPLError(CE_Warning, CPLE_AppDefined,
+                 "NOTE: The observer location falls outside of the DEM area");
+        //ABELL - Make sure observer Z is specified.
```

Comment:
some check to add here ?

Reply:
I'm not sure. Normally the Z is zero by default and perhaps this is still fine.